base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/configuration/ConfigInfo.java

```diff
@@ -120,10 +120,11 @@ public String synoptic() {
      * @return The list of config names
      */
     public static Collection<String> names(Collection<ConfigInfo> infos) {
-        if (infos == null) {
+        if (infos.isEmpty()) {
```

Comment:
Though I agree we should check for emptiness and ideally wouldn't need null checks, I do think we should not remove null checks when we can't guarantee how `infos` is made. I suggest `if (infos == null || infos.isEmpty())` as the `||` is cutting so if infos is null then the check `infos.isEmpty()` will not be performed. Please do similar for the other places that we have removed the null check.